src/fsharp/FSharp.Core/seq.fs

```diff
@@ -1486,26 +1491,26 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Unfold")>]
         let unfold (generator:'State->option<'T * 'State>) (state:'State) : seq<'T> =
-            SeqComposer.Helpers.upcastEnumerable (new SeqComposer.Unfold.Enumerable<'T,'T,'State>(generator, state, SeqComposer.IdentityFactory.IdentityFactory))
+            Composer.Helpers.upcastEnumerable (new Composer.Unfold.Enumerable<'T,'T,'State>(generator, state, Composer.IdentityFactory.IdentityFactory))
 
         [<CompiledName("Empty")>]
         let empty<'T> = (EmptyEnumerable :> seq<'T>)
 
         [<CompiledName("InitializeInfinite")>]
         let initInfinite<'T> (f:int->'T) : IEnumerable<'T> =
-            SeqComposer.Helpers.upcastEnumerable (new SeqComposer.Init.EnumerableDecider<'T>(Nullable (), f))
+            Composer.Helpers.upcastEnumerable (new Composer.Init.EnumerableDecider<'T>(Nullable (), f))
 
         [<CompiledName("Initialize")>]
         let init<'T> (count:int) (f:int->'T) : IEnumerable<'T> =
             if count < 0 then invalidArgInputMustBeNonNegative "count" count
             elif count = 0 then empty else
-            SeqComposer.Helpers.upcastEnumerable (new SeqComposer.Init.EnumerableDecider<'T>(Nullable count, f))
+            Composer.Helpers.upcastEnumerable (new Composer.Init.EnumerableDecider<'T>(Nullable count, f))
 
         [<CompiledName("Iterate")>]
         let iter f (source : seq<'T>) =
             source
             |> foreach (fun _ ->
-                    { new SeqComposer.SeqConsumer<'T,'T> () with
+                    { new Composer.Internal.Consumer<'T,'T> () with
                         override this.ProcessNext value =
                             f value
                             Unchecked.defaultof<bool> })
```

Comment:
Why `Unchecked.defaultof<bool>` instead of `false`? Performance reasons? Does `Unchecked.defaultof<bool>` cause one less IL instruction to be generated compared to `false`, and is there a significant performance gain to counterbalance the cost of having less-readable code? It's not a huge readability cost in this case, but it's non-zero. I'd like to know that there's at least a somewhat significant performance gain (i.e., more than 1%) to justify the cost here (and in the 21 other places where `Unchecked.defaultof<bool>` is used).